docs/source/en/tasks/asr.mdx

```diff
@@ -212,15 +264,89 @@ At this point, only three steps remain:
 ...     eval_dataset=encoded_minds["test"],
 ...     tokenizer=processor.feature_extractor,
 ...     data_collator=data_collator,
+...     compute_metrics=compute_metrics,
 ... )
 
 >>> trainer.train()
 ```
+
+Once training is completed, share your model to the Hub with the [`~transformers.Trainer.push_to_hub`] method so everyone can use your model:
+
+```py
+>>> trainer.push_to_hub()
+```
 </pt>
 </frameworkcontent>
 
 <Tip>
 
-For a more in-depth example of how to fine-tune a model for automatic speech recognition, take a look at this blog [post](https://huggingface.co/blog/fine-tune-wav2vec2-english) for English ASR and this [post](https://huggingface.co/blog/fine-tune-xlsr-wav2vec2) for multilingual ASR.
+For a more in-depth example of how to finetune a model for automatic speech recognition, take a look at this blog [post](https://huggingface.co/blog/fine-tune-wav2vec2-english) for English ASR and this [post](https://huggingface.co/blog/fine-tune-xlsr-wav2vec2) for multilingual ASR.
+
+</Tip>
+
+## Inference
+
+Great, now that you've finetuned a model, you can use it for inference!
+
+Load an audio file you'd like to run inference on. Remember to resample the sampling rate of the audio file to match the sampling rate of the model if you need to!
+
+```py
+>>> from datasets import load_dataset, Audio
+
+>>> dataset = load_dataset("PolyAI/minds14", "en-US", split="train")
+>>> dataset = dataset.cast_column("audio", Audio(sampling_rate=16000))
+>>> sampling_rate = dataset.features["audio"].sampling_rate
+>>> audio_file = dataset[0]["audio"]["path"]
+```
+
+The simplest way to try out your finetuned model for inference is to use it in a [`pipeline`]. Instantiate a `pipeline` for automatic speech recognition with your model, and pass your audio file to it:
+
+<Tip>
+
+The results aren't very good because the model was only trained on a few examples to quickly demonstrate the process. Try finetuning your own model on the full dataset to get better results!
+
+</Tip>
+
+```py
+>>> from transformers import pipeline
+
+>>> transcriber = pipeline("automatic-speech-recognition", model="stevhliu/my_awesome_asr_minds_model")
+>>> transcriber(audio_file)
+{'text': "ALDCBC</s>COCB</s>DCBDBCBRBCDCTCRWKDBK<s></s>KODT</s>C</s>T</s>C</s>OD<s>B</s>BCTOR</s>AKDOBODTCTEDQOATDVDKCDSKL<unk>KEKEPKNPKPADSRDSCLCQ<unk>ED<s>BDXAXDP<unk>KQDPWDKLEPRPKDTDRDPDNFRTADPLDSRDPDKDX<unk>XLXCXLCL</s>LCXLCLDT</s>TBLDLBLTLCB</s>BXTLTBLDTM</s>BLBLCLBLCLDT</s>LXCDTB</s>L</s>T</s>LTDK</s>TDTDTLOKDADPDSDT</s><unk>RDQTDDADSNDN</s>KD</s>T'DPNVPHKDPDRDRKRKD</s>PKDCQS<unk>X<unk>LTAQDXLXCTBCLBTQBCXBDQDTXM<unk>LCXL</s>CXT"}
```

Comment:
Maybe we can train on a few hundred samples to get a somewhat reasonable transcription? Or is the point of this .md purely to demonstrate **how** to fine-tune (rather than with any care to performance), meaning we're happy with mumbo jumbo here?

Reply:
I think I may have run out of memory in Colab when I tried with more samples, but I'll try again with a few hundred samples to see if we can do better than this gibberish.

Reply:
Note that when you download a dataset using `train[:20]` you actually download the entire dataset and then slice the first 20 samples -> so in terms of disk space using a couple of hundred samples will be equivalent to just 20.

For training, you might be getting OOM's because there are very long audio samples in your dataset. The self-attention mechanism memory scales with sequence length squared: $\text{memory} \propto \text{seq len}^2$. So there could be long samples > 20s that are causing OOM's! We usually filter such samples by a 20s max length criteria: 
https://github.com/huggingface/transformers/blob/a0f867430347bcf939f71d186409b9ca138c3b34/examples/pytorch/speech-recognition/run_speech_recognition_ctc.py#L634-L642

Reply:
Finally trained a better model that doesn't return mumbo jumbo and added an important note on making sure the dataset text matches the casing of the tokenizer!

Reply:
Nice! Glad to hear it's working 🗣